src/pages/Order/hooks/useReceivers.ts

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import { receiverListSchema, receiverSchema, type Receiver, type ReceiverError } from '../schemas/receiverSchema';
+
+export const useReceivers = (initialReceivers: Receiver[]) => {
+  const [receivers, setReceivers] = useState<Receiver[]>(initialReceivers);
+  const [errors, setErrors] = useState<ReceiverError[]>([]);
+
+  const addReceiver = () => {
+    if (receivers.length >= 10) return;
+    setReceivers((prev) => [...prev, { name: '', phone: '', quantity: 1 }]);
+  };
+
+  const removeReceiver = (index: number) => {
+    setReceivers((prev) => prev.filter((_, i) => i !== index));
+  };
+
+  const updateReceiver = (index: number, field: keyof Receiver, value: string) => {
+    const updated = [...receivers];
+    const current = updated[index];
+
+    updated[index] = {
+      ...current,
+      [field]: field === 'quantity' ? Number(value) : value,
+    };
+
+    setReceivers(updated);
+    validate(index, updated[index]);
+  };
+
+  const validate = (index: number, receiver: Receiver) => {
+    const result = receiverSchema.safeParse(receiver);
+    const newErrors = [...errors];
+    if (!result.success) {
+      const fieldErrors: ReceiverError = {};
+      for (const issue of result.error.issues) {
+        if (issue.path[0]) {
+          fieldErrors[issue.path[0] as keyof Receiver] = issue.message;
+        }
+      }
+      newErrors[index] = fieldErrors;
+    } else {
+      newErrors[index] = {};
+    }
+    setErrors(newErrors);
+  };
```

Comment:
```suggestion
const validate = (index: number, receiver: Receiver) => {
  const validationResult = receiverSchema.safeParse(receiver);
  
  const fieldErrors = validationResult.success 
    ? {} 
    : validationResult.error.issues.reduce((acc: ReceiverError, issue) => {
        const fieldName = issue.path[0] as keyof Receiver;
        return fieldName ? { ...acc, [fieldName]: issue.message } : acc;
      }, {});

  setErrors(prevErrors => 
    prevErrors.map((error, idx) => idx === index ? fieldErrors : error)
  );
};
```

이렇게 reduce를 활용할 수 있답니다 ㅎㅎ